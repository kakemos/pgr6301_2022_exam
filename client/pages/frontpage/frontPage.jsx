import React from "react";
import { PreviewAllArticles } from "../news/previewAllArticles";

export function FrontPage() {
  return (
    <>
      <div>
        <h1 id={"frontpage-title"}>Everyday News!</h1>
      </div>
      <div>
        <PreviewAllArticles />
      </div>
    </>
  );
}
