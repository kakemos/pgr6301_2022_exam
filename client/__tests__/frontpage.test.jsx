import {BrowserRouter} from "react-router-dom";
import {FrontPage} from "../pages/frontpage/frontPage";
import ReactDOM from "react-dom";
import React from "react";

describe("front page component", () => {
    it("shows headline", () => {
        const domElement = document.createElement("div");
        ReactDOM.render(
            <BrowserRouter>
                <FrontPage/>
            </BrowserRouter>,
            domElement);
        expect(domElement.querySelector("#frontpage-title").innerHTML).toEqual("Everyday News!")
        expect(domElement.innerHTML).toMatchSnapshot();
    })
})