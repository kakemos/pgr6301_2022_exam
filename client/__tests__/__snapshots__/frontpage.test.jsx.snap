// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`front page component shows headline 1`] = `"<div><h1 id=\\"frontpage-title\\">Everyday News!</h1></div><div><div>Loading...</div></div>"`;
